modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1536,6 +1583,355 @@ MULC_SIMD(float, float)
 
 #undef MULC_SIMD
 
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC, typename DST>
+CV_ALWAYS_INLINE
+typename std::enable_if<(std::is_same<DST, ushort>::value ||
+                         std::is_same<DST, short>::value), int>::type
+divc_simd_common_impl(scale_tag_t s_tag, const SRC* inx, DST* outx,
+                      const v_float32& v_scalar, const v_float32& v_scale,
+                      const int length)
+{
+    constexpr int nlanes = vector_type_of_t<DST>::nlanes;
+
+    v_float32 v_zero = vx_setzero_f32();
+    v_float32 v_mask = (v_scalar == v_zero);
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = vg_load_f32(inx);
+            v_float32 a2 = vg_load_f32(&inx[nlanes/2]);
+
+            v_store_i16(outx, v_round(v_select(v_mask, v_zero, div_op(s_tag, a1, v_scalar, v_scale))),
+                              v_round(v_select(v_mask, v_zero, div_op(s_tag, a2, v_scalar, v_scale))));
+        }
+
+        if (x < length)
+        {
+            x = length - nlanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC>
+CV_ALWAYS_INLINE int divc_simd_common_impl(scale_tag_t s_tag, const SRC* inx,
+                                           uchar* outx, const v_float32& v_scalar,
+                                           const v_float32& v_scale, const int length)
+{
+    constexpr int nlanes = v_uint8::nlanes;
+
+    v_float32 v_zero = vx_setzero_f32();
+    v_float32 v_mask = (v_scalar == v_zero);
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = vg_load_f32(inx);
+            v_float32 a2 = vg_load_f32(&inx[nlanes/4]);
+            v_float32 a3 = vg_load_f32(&inx[nlanes/2]);
+            v_float32 a4 = vg_load_f32(&inx[3 * nlanes/4]);
+
```

Comment:
but why we make 4 loads in case of `uchar`  output format?
Is it true that a1,a2,... contains intersected data and we just wasting or cutting  these intersected values in packX ?

Reply:
> but why we make 4 loads in case of `uchar` output format?

Because 4  float will be packed to uchar. So they will be transformed from 32-bits to 8-bits numbers.

For example I choose 128-bits SIMD vector. 
It can contain 4 x floats(4x32=128) or 16 x uchars(16x8=128). We load 4 v_float32 vectors. Each vector contains 4 float numbers. In total we have 16 float numbers in 4 SIMD vectors after loading. Then we transform 16 float numbers to 16 uchar numbers(get 8-bits x 16 numbers=128 bits) and put it to 128-bit SIMD vector.

Reply:
> Is it true that a1,a2,... contains intersected data and we just wasting or cutting these intersected values in packX ?

a1, a2, a3 and a4 **_doesn't contain_** intersected data.